openlibrary/templates/books/edition-sort.html

```diff
@@ -60,61 +54,58 @@
                 $if book.languages:
                     in $', '.join(l.name for l in book.languages)
             $if book.edition_name:
-                -
-                $book.edition_name
+                - $book.edition_name
             </div>
         </div>
     </td>
 
-    $if 'printdisabled' in collection or 'lendinglibrary' in collection:
-        <td class="icon read">
-        <div class="bbbb"></div>
+    $ availability_class = 'zzzz' if (not availability or availability.get('status') == 'error') else 'aaaa' if (availability.get('status') == 'open') else 'bbbb' if (availability.get('status') == 'borrow_available') else 'cccc' if (availability.get('status') == 'borrow_unavailable') else 'dddd'
+    $ active = False if availability_class == 'zzzz' else True
+
+    <td class="icon read $('inact' if not active else '')">
+      <div class="$(availability_class)"></div>
         <div class="links">
+          $if active:
             <ul>
-            <li><a href="$book.url('/daisy')" class="plain"><span class="underline">DAISY</span> <img src="/images/icons/icon-encrypto-sm.png" width="11" height="14" alt="Protected DAISY"/></a></li>
+              $if availability.get('status') == 'open':
+                <li class="read-option"><strong><a href="$viewbook.replace('XXX', book.ocaid)" title="Read a scanned version in BookReader" class="read-btn">Read eBook</a></strong></li>
+
+              $elif availability.get('status') in ['borrow_available', 'borrow_unavailable']:
+                <li class="read-option">$:macros.LoanStatus(book, ctx.user)</li>
+
+              <li class="read-option">
+                <a href="$book.url('/daisy')" class="plain print-disabled-download">
+                  <span class="underline">DAISY for print-disabled</span>
+                  <img src="/images/icons/icon-encrypto-sm.png" class="daisy-lock" width="8" height="12" alt="Download ebook for print-disabled (DAISY)"/>
+                </a>
+              </li>
+
+              $if availability.get('status') == 'open':
+                <hr>
+                <ul>
+                  <li><a href="$downloadbook.replace('XXX', book.ocaid)" title="Download a PDF">PDF</a></li>
+                  <li><a href="$epub.replace('XXX', book.ocaid)" title="Download an ePub">ePub</a></li>
+                  <li><a href="$mobi.replace('XXX', book.ocaid)" title="Download a MOBI file from Internet Archive">MOBI</a></li>
+                  <li><a href="$djvutxt.replace('XXX', book.ocaid)" title="Download a plain text version">Plain text</a></li>
+                </ul>
             </ul>
         </div>
-        </td>
-    $elif book.ocaid:
-        <td class="icon read">
-        <div class="aaaa"></div>
+    </td>
+
+    <td class="icon locate inact">
+      <div class="zzzz"></div>
```

Comment:
SUGGESTION: Use a more meaningful class name than `zzzz`.

Reply:
It's used as the sorting selector. I agree with the suggestion though am going to punt on an alternative for now (it impacts a lot of companion js code)